doc/template_definition.md

```diff
@@ -81,8 +124,79 @@ method.getBody().insertBegin(injectedCode);
 
 ```
 
-Kinds of templating
--------------------
+Definition of templates using `PatternBuilder`
+--------------------------------------------------
+
+The body of method `CheckBoundTemplate#statement` below defines a Spoon template. 
+
+```java
+public class CheckBoundTemplate /*it doesn't matter what it extends or implements*/ {
+	// it doesn't matter what other type members are in template class
+	// it doesn't matter what is the name of the method whose body will be used as template
+	public void statement(Collection<?> _col_) {
+		if (_col_.size() > 10)
+			throw new OutOfBoundException();
+	}
+}
+```
+
+The code, which creates a Spoon template using `PatternBuilder` looks like this:
+```java
+Pattern t = PatternBuilder.create(factory,
+	//defines template class.
+	CheckBoundTemplate.class, 
+	//defines which part of template class will be used as template model
+	model -> model.setBodyOfMethod("statement"))
```

Comment:
Could it be a specific statement of the method?

Reply:
Actually not, but if you see this use case interesting, then it is no problem to add new method into that builder.

Reply:
I just wanted to know what kind of filter could be done here, but I get an idea looking on `TemplateModelBuilder` :)